### ISSUE TITLE
chore: add Biome for linting and formatting configuration & fix lint/format errors

### ISSUE THREAD

**Issue by Daniel-Boll** [basementdevs/scylla-studio#25]
**Description**:
We need to integrate [Biome](https://biomejs.dev/) as the main linting and formatting tool for the project to maintain code quality and consistency. This task involves setting up Biome and fixing all the lint errors and formatting issues it identifies in the codebase.

**Tasks**:

1. Add `biome` configuration for linting and formatting.
   - Install `biome` in the project.
   - Configure linting rules according to the project's standards.
   - Configure formatting options (if needed).

2. Run `biome` to check for linting and formatting issues.
   - Ensure all files in the project are checked.
   - Investigate and fix every lint and formatting error reported.

3. Set up linting and formatting to run automatically via:
   - Pre-commit hooks (using `husky` or any existing tool) [In doubt ask @gvieira18].
   - CI pipeline (if applicable).

4. Ensure documentation is updated (if necessary):
   - Add instructions for running `biome` in `README.md` or any relevant documentation file.

**Acceptance Criteria**:
- `biome` is configured in the project.
- All existing lint errors and formatting issues are fixed.
- Linting and formatting run as part of pre-commit hooks and/or CI pipeline.

**Comment by void-hr** [comment]
@Daniel-Boll, can you assign me this task?

**Comment by Daniel-Boll** [comment]
@void-hr you are good to go, feel free to open a PR whenever this is ready, what counts is the PR merged itself.

**Comment by gvieira18** [comment]
NextJs already has its own config for linting (`npm run lint`), and it doesn't look friendly for [out-of-the-box integration at the moment](https://github.com/vercel/next.js/discussions/59347?sort=new).

Maybe we can try to [migrate the config from eslint](https://biomejs.dev/guides/migrate-eslint-prettier/), but it might be more complicated than just avoiding all the previous config and creating a new one.

For git-hooks, we already use lefthook on [tbp-extension](https://github.com/basementdevs/tbp-extension), and it works well.

**Comment by Daniel-Boll** [comment]
Okay, I think we can stick with the following:

- [ ] lint: eslint (Next's default)
- [ ] formatter: biome
- [ ] git-hooks: lefthook

What do you think?

**Comment by void-hr** [comment]
I agree with the approach to stick with ESLint for linting, as it's deeply integrated with Next.js and would save us from unnecessary complexity. For formatting, I think Biome can easily be introduced as a standalone formatter without interfering with the current linting setup.
 
If anyone has additional suggestions or concerns regarding this approach, feel free to share them, and I’ll adjust the plan accordingly.

**Comment by void-hr** [comment]
@gvieira18 @Daniel-Boll, any other suggestions, should I start to make changes ?

**Comment by gvieira18** [comment]
> @gvieira18 @Daniel-Boll, any other suggestions, should I start to make changes ?

I think you can do the tasks.

**Comment by Daniel-Boll** [comment]
> Referring to biome config as used in [tbp-extension](https://github.com/basementdevs/tbp-extension/blob/develop/biome.jsonc), any additional config you want me to add ?

No, that sounds perfect already.
